src/region.js

```diff
@@ -269,7 +275,11 @@ const Region = MarionetteObject.extend({
     delete this.currentView;
 
     if (!view._isDestroyed) {
-      this._removeView(view, shouldDestroy);
+      if (shouldDestroy) {
+        this.removeView(view);
+      } else {
+        this._detachView(view);
```

Comment:
I added tests, checking whether `removeView` is called depending on `preventDestroy` and after `detachView`